auth_backend/backends.py

```diff
@@ -6,7 +6,17 @@
 
 class KagisoBackend(ModelBackend):
 
-    def authenticate(self, email, password, **kwargs):
+    # HACK: username is actually email, but Django passes in keyword args
```

Comment:
django determines what backend to use based on the names of your parameters

so in competition_builder i created my own AdminBackend which uses (username, password) and my CASBackend that uses (email, password)

it works beautifully